### PR TITLE
Home screen UI

### PR DESCRIPTION
**Pull request by divyeshp07**
Added home screen UI.
Expanded color palette in the core.
Edited existing app color extension.
Renamed "rounded button" to "app icon button" in the core.
Modified dark and light themes.
Added a tab bar widget.

### CODE REVIEW

File: lib/features/authentication/view/widgets/tabbar_widget.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        separatorBuilder: (context, index) =>
+            SizedBox(width: context.spaces.space_25),
+        scrollDirection: Axis.horizontal,
+        itemCount: catogryItems.length,
+        itemBuilder: (context, index) {
+          return GestureDetector(
+            onTap: onTap,
+            child: Container(
+              margin: EdgeInsets.all(context.spaces.space_100),
+              // width: apptheme.spaces.space_700,
```
Comment (Sireen-Official) [inline lib/features/authentication/view/widgets/tabbar_widget.dart:33]: Remove this line

**Review by Sireen-Official — CHANGES_REQUESTED** [review]
Remove comments, only use comments for documentation

**Review by thadathilsinan — APPROVED** [review]

File: lib/features/authentication/view/widgets/tabbar_widget.dart
Comment (Sireen-Official) [inline lib/features/authentication/view/widgets/tabbar_widget.dart:1]: Use the tab bar widget here

**Review by Sireen-Official — CHANGES_REQUESTED** [review]

File: lib/features/authentication/view/pages/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<String> categoryList =
+        ref.read(filterSortConstantsProvider).productType;
+
+    return DefaultTabController(
+      length: 5,
```
Comment (Sireen-Official) [inline lib/features/authentication/view/pages/home_page.dart:44]: Unnecessary default tab controller, remove it.

File: lib/features/authentication/view/pages/home_page.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ],
+            body: TabBarView(
+              controller: tabController,
+              children: [
+                ListView.builder(
```
Comment (Sireen-Official) [inline lib/features/authentication/view/pages/home_page.dart:86]: Extract this listview builder to one widget

**Review by Sireen-Official — CHANGES_REQUESTED** [review]

**Review by Sireen-Official — APPROVED** [review]

**Pull request merged**